### PR TITLE
Added renv, eliminated R CMD Notes

### PR DESCRIPTION
**Pull request by ms130**
Travis CI doesn't seem to work anymore, so replaced it with GitHub Actions. Remove Travis badge. Fixed R CMD Notes by adding `knitr` call before `opts_chunk`. Added `renv` for local package management, as it makes life easier.

### CODE REVIEW

**Comment by BrydenFrancesca** [comment]
@ms130 this looks like a good addition to me! One note; the usual workflow files aren't present in the .github repo (I'd expect a yaml file). Not sure if you've set it up in another way to how I'm used to (I use `usethis::use_github_action_check_standard()`), or if it's a problem.

File: .github/workflows/check-release.yaml [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+  push:
+    branches: [main, master, dev, development]
+  pull_request:
+    branches: [main, master, dev, development]
+
+name: R-CMD-check
+
+jobs:
+  R-CMD-check:
+    runs-on: ubuntu-latest
```
Comment (BrydenFrancesca) [inline .github/workflows/check-release.yaml]: Do we want the CMD check to run on windows too?

**Review by BrydenFrancesca — APPROVED** [review]
Ignore previous comment; it is just because you've called the Github Action slightly differently than I'm used to!

File: .github/workflows/check-release.yaml (hunk 1 above)
Comment (ms130) [reply .github/workflows/check-release.yaml]: Good idea, have added a more robust workflow checking Windows too

**Pull request merged**